easybuild/easyblocks/generic/cargo.py

```diff
@@ -270,11 +283,16 @@ def extract_step(self):
         for src in self.src:
             # Check for git crates, `git_key` will be set to a true-ish value for those
             try:
-                crate_name, _, git_repo, rev = vendor_crates[src['name']]
+                crate = vendor_crates[src['name']]
+                if len(crate) == 5:
+                    crate_name, _, git_repo, rev, branch = crate
+                else:
+                    crate_name, _, git_repo, rev = crate
+                    branch = self.BRANCH_MISSING
```

Comment:
Maybe we can simplify this to if we can spare the `BRANCH_MISSING` state
```suggestion
                crate = vendor_crates[src['name']]
                crate_name, _, *git_key = crate
```

Reply:
I didn't know you can have `*git_key` here, thanks! But that's too loose. Rather:
```suggestion
                crate_name, _, repo, rev, *branch = crate
```

Because we do need the repo and revision anyway and want it to fail for `(name, version)` tuples. Only downside is the branch will be `[]` which we should change to `None` below.